### PR TITLE
Port to ipaddress package

### PR DESCRIPTION
**Pull request by NotAFile**
* Port to ipaddress package
* Add tests for networkdict

### CODE REVIEW

File: piqueserver/networkdict.py [hunk 1]
```diff
@@ -7,19 +7,18 @@ def get_network(cidr):
     try:
         return CACHE[cidr]
     except KeyError:
-        network = IPNetwork(cidr)
+        network = ip_network(unicode(cidr))
```
Comment (NotAFile) [inline piqueserver/networkdict.py]: This breaks on py3

File: piqueserver/networkdict.py [hunk 2]
```diff
@@ -7,19 +7,18 @@ def get_network(cidr):
...
         CACHE[cidr] = network
         return network
 
-
 def get_cidr(network):
     # TODO: why are we accessing a protected attribute?
     #       does this work?
+    # testing for IPv4?
```
Comment (NotAFile) [inline piqueserver/networkdict.py]: Enet only supports IPv4. This is testing if the network is a `/32`, i.e. is a single IP address.

File: piqueserver/networkdict.py [hunk 3]
```diff
@@ -7,19 +7,18 @@ def get_network(cidr):
     try:
```
Comment (NotAFile) [inline piqueserver/networkdict.py]: Do we even need get_network? Why is it caching stuff?

File: piqueserver/networkdict.py [hunk 4]
```diff
@@ -54,7 +53,7 @@ def __getitem__(self, key):
         return self.get_entry(key)[1]
 
     def get_entry(self, key):
-        ip = get_network(key)
+        ip = ip_address(unicode(key))
```
Comment (NotAFile) [inline piqueserver/networkdict.py]: same as above

File: piqueserver/networkdict.py (hunk 1 above)
Comment (godwhoa) [reply piqueserver/networkdict.py]: How do I make it compatible with both python 2/3?

Comment (NotAFile) [reply piqueserver/networkdict.py]: It requires a unicode object, I guess? There's a few options, but it's not really easy. You could require the callee to give a valid unicode string object, alias unicode to str if unicode is not available or convert it to unicode only if it's a bytes object. We've only had this issue in reverse before.

File: piqueserver/networkdict.py (hunk 3 above)
Comment (godwhoa) [reply piqueserver/networkdict.py]: ip_network isn't an expensive call, we can remove it.

Comment (NotAFile) [reply piqueserver/networkdict.py]: What I thought might be going on is that it is constanty recreating these when syncing bans

File: piqueserver/networkdict.py [hunk 5]
```diff
@@ -1,20 +1,9 @@
...
-    except KeyError:
-        network = ip_network(unicode(cidr))
-        CACHE[cidr] = network
-        return network
-
 def get_cidr(network):
     # TODO: why are we accessing a protected attribute?
     #       does this work?
-    # testing for IPv4?
     if network._prefixlen == 32:  # pylint: disable=protected-access
```
Comment (NotAFile) [inline piqueserver/networkdict.py]: network._prefixlen shouldn't exist in ipaddress AFAIK

**Comment by NotAFile** [comment]
can you add a comment why strict=False is needed? Otherwise someone is going to be hella confused in a while.

**Pull request merged**